util-core/src/test/scala/com/twitter/concurrent/exp/AsyncStreamTest.scala

```diff
@@ -192,6 +192,37 @@ class AsyncStreamTest extends FunSuite with GeneratorDrivenPropertyChecks {
     }
   }
 
+  test("scanLeft is eager") {
+    val first = new Promise[Unit]
+
+    var stream = AsyncStream.fromFuture(first map (_ => 2)).scanLeft(1)(_ + _)
```

Comment:
ffti: the style is now to avoid infix except for symbolic operators, so this should be `first.map(_ => 2)`